demonstrations/tutorial_liealgebra.py

```diff
@@ -0,0 +1,392 @@
+# 
+# Dynamical Lie Algebras
+# ~~~~~~~~~~~~~~~~~~~~~~
+#
+# A different way of looking at this is taking a set of generators :math:`\{iG_j\}` and asking what kind of 
+# unitary evolutions they can generate. This naturally introduces the so-called Dynamical Lie Algebra (DLA),
+# originally proposed in quantum optimal control theory and recently re-emerging in the quantum computing literature.
+# The DLA :math:`i\mathfrak{g}` is given by all possible nested commutators between the generators :math:`\{iG_j\}`,
+# until no new and linearly independent skew-Hermitian operator is generated. This is called the Lie-closure and is
+# written like
+#
+# .. math:: i \mathfrak{g} = \langle iG_1, iG_2, iG_3,.. \rangle_\text{Lie}.
+#
+# Let us do a quick example and compute the Lie closure of :math:`\{iX, iY\}` (more examples later).
+
+print(qml.commutator(1j * X(0), 1j * Y(0)))
+
+##############################################################################
+# We know that the commutator between :math:`iX` and :math:`iY` yields a new operator :math:`\propto iZ`.
+# Note that we do not care for scalar coefficients, just the operators (technically, we care for linear independence, and :math:`2i Z` is of course linearly dependent on :math:`iZ`).
+# So we add :math:`iZ` to our list of operators and continue to take commutators between them.
+
+list_ops = [1j * X(0), 1j * Y(0), 1j * Z(0)]
+for op1 in list_ops:
+    for op2 in list_ops:
+        print(qml.commutator(op1, op2))
+
+##############################################################################
+# Since no new operators have been created we know the lie closure is complete and our dynamical Lie algebra
+# is :math:`\langle\{iX, iY\}\rangle_\text{Lie} = \{iX, iY, iZ\}( = \mathfrak{su}(2))`.
+#
+# On one hand, the Lie closure ensures that the DLA is closed under commutation.
+# But you can also think of the Lie closure as filling the missing operators to describe the possible dynamics in terms of its Lie algebra.
+# Let us stick to the example above and imagine for a second that we dont take the Lie closure but just take the two generators :math:`\{iX, iY\}`.
+# These two generators suffice for universality (for a single qubit) in that we can write any evolution in the Dynamical Lie Group :math:`SU(2)` as a finite product of 
+# these :math:`X` and :math:`Y` rotations :math:`e^{-i \phi X}` and :math:`e^{-i \phi Y}`. 
+# For example, let us write a Pauli-Z rotation at non-trivial angle :math:`0.5` as a product of them.
+
+U_target = qml.matrix(qml.RZ(-0.5, 0))
+decomp = qml.ops.one_qubit_decomposition(U_target, 0, rotations="XYX")
+print(decomp)
+
+##############################################################################
+# We can check that this is indeed a valid decomposition by computing the trace distance to the target.
+
+U = qml.matrix(decomp[0])
+U = U @ qml.matrix(decomp[1])
+U = U @ qml.matrix(decomp[2])
```

Comment:
```suggestion
U = qml.prod(*decomp).matrix()
```
Another optional suggestion.